webhook/resources/engineimage/mutator.go

```diff
@@ -0,0 +1,62 @@
+package engineimage
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/longhorn/longhorn-manager/datastore"
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+	"github.com/longhorn/longhorn-manager/types"
+	"github.com/longhorn/longhorn-manager/webhook/admission"
+)
+
+type engineImageMutator struct {
+	admission.DefaultMutator
+	ds *datastore.DataStore
+}
+
+func NewMutator(ds *datastore.DataStore) admission.Mutator {
+	return &engineImageMutator{ds: ds}
+}
+
+func (e *engineImageMutator) Resource() admission.Resource {
+	return admission.Resource{
+		Name:       "engineimages",
+		Scope:      admissionregv1.NamespacedScope,
+		APIGroup:   longhorn.SchemeGroupVersion.Group,
+		APIVersion: longhorn.SchemeGroupVersion.Version,
+		ObjectType: &longhorn.EngineImage{},
+		OperationTypes: []admissionregv1.OperationType{
+			admissionregv1.Create,
+		},
+	}
+}
+
+func (e *engineImageMutator) Create(request *admission.Request, newObj runtime.Object) (admission.PatchOps, error) {
+	var patchOps admission.PatchOps
+
+	engineImage := newObj.(*longhorn.EngineImage)
+
+	image := strings.TrimSpace(engineImage.Spec.Image)
+	if image != engineImage.Spec.Image {
+		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/image", "value": "%s"}`, image))
+	}
+
+	name := types.GetEngineImageChecksumName(image)
+	if name != engineImage.Name {
+		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/metadata/name", "value": "%s"}`, name))
+	}
+
+	labels := types.GetEngineImageLabels(name)
```

Comment:
Can we allow users to set the customized labels during this CR creation?

Reply:
Then will op `replace` overwrite the labels customized by users themselves?

Reply:
Yes. This will overwrite the labels set by users.
Do you think we need to merge the labels set by `GetEngineImageLabels` and users?
Originally, the labels are from `GetEngineImageLabels`, correct?

Reply:
IMO, Longhorn just needs to **add and correct** the labels for the **specific** keys.